lib/traject/macros/met_csv.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module Macros
+  # Macros for extracting values from CSV rows
+  module MetCsv
+    # @param header_or_index [String] the field header or index to accumulate
+    def column(header_or_index, options = {})
+      lambda do |row, accumulator, _context|
+        result = Array(row[header_or_index])
+        result = result.map(&:strip) if options.key?(:trim)
+        accumulator.concat(result)
+      end
+    end
+
+    def generate_creator
+      lambda do |row, accumulator, _context|
+        display_name = row['Artist Display Name']
+        suffix = row['Artist Suffix']
+        begin_date = row['Artist Begin Date']
+        end_date = row['Artist End Date']
+        role = row['Artist Role']
+        display_bio = row['Artist Display Bio']
+
+        if display_name.present?
+          display_pattern = display_name
+        end
+
+        if suffix.present?
+          display_pattern = "#{display_pattern}, #{suffix}"
+        end
+
+        if begin_date.present?
+          display_pattern = "#{display_pattern}, #{begin_date}"
+        end
+
+        if end_date.present?
+          display_pattern = "#{display_pattern} - #{end_date}"
+        end
+
+        if role.present?
```

Comment:
Do we need to worry about what happens if some of these values are missing or blank? It seems like the string formatting won't be quite right without it, yeah?